src/nvapi_disp.cpp

```diff
@@ -96,15 +89,85 @@ extern "C" {
         if (output == nullptr)
             return InvalidArgument(n);
 
-        switch (pHdrColorData->version) {
-            case NV_HDR_COLOR_DATA_VER1:
-            case NV_HDR_COLOR_DATA_VER2:
-                break;
-            default:
-                return IncompatibleStructVersion(n);
+        auto interop = nvapiAdapterRegistry->GetInteropFactory();
+        if (interop == nullptr)
+            return NoImplementation(str::format(n, " (0x", std::hex, displayId, ")"));
+
+        // V2 is based on V1.
+        if (pHdrColorData->version == NV_HDR_COLOR_DATA_VER1 || pHdrColorData->version == NV_HDR_COLOR_DATA_VER2) {
+            auto pHDRColorDataV1 = reinterpret_cast<NV_HDR_COLOR_DATA_V1*>(pHdrColorData);
+            if (pHDRColorDataV1->cmd == NV_HDR_CMD_GET) {
+                DXGI_COLOR_SPACE_TYPE colorspace;
+                DXGI_HDR_METADATA_HDR10 metadata;
+                if (interop->GetGlobalHDRState(&colorspace, &metadata) == S_OK) {
+                    pHDRColorDataV1->hdrMode = colorspace != DXGI_COLOR_SPACE_RGB_FULL_G22_NONE_P709 ? NV_HDR_MODE_UHDA : NV_HDR_MODE_OFF;
```

Comment:
This looks wrong. I've modified dozens of games that use NvAPI for HDR, not a single one has ever used `NV_HDR_MODE_UHDA`. They all use `NV_HDR_MODE_UHDA_PASSTHROUGH`.

`NV_HDR_MODE_UHDA` would be scRGB if even a single game ever used it. `NV_HDR_MODE_UHDA_PASSTHROUGH` is HDR10. Despite NVIDIA's docs saying it's "experimental", it is in fact the only format any shipping game uses aside from Mass Effect: Andromeda (uses `NV_HDR_MODE_DOLBY_VISION` as an alternative).

Reply:
This is the "Get" mode, but yeah. My thinking here was that if the global HDR toggle in Windows is on, it'd be scRGB here. But maybe it should be UHDA_PASSTHRU. I will check native behaviour.